scene/resources/curve.cpp

```diff
@@ -1630,114 +1698,150 @@ Vector3 Curve3D::sample_baked(real_t p_offset, bool p_cubic) const {
 	}
 }
 
-real_t Curve3D::sample_baked_tilt(real_t p_offset) const {
-	if (baked_cache_dirty) {
-		_bake();
-	}
-
-	// Validate: Curve may not have baked tilts.
-	int pc = baked_tilt_cache.size();
-	ERR_FAIL_COND_V_MSG(pc == 0, 0, "No tilts in Curve3D.");
+real_t Curve3D::_sample_baked_tilt(Interval p_interval) const {
+	// Assuming that p_interval is valid.
+	ERR_FAIL_INDEX_V_MSG(p_interval.idx, baked_tilt_cache.size(), 0.0, "Invalid interval");
 
-	if (pc == 1) {
-		return baked_tilt_cache.get(0);
-	}
+	int idx = p_interval.idx;
+	real_t frac = p_interval.frac;
 
 	const real_t *r = baked_tilt_cache.ptr();
 
-	if (p_offset < 0) {
-		return r[0];
+	return Math::lerp(r[idx], r[idx + 1], frac);
+}
+
+Basis Curve3D::_sample_posture(Interval p_interval, bool p_apply_tilt) const {
+	// Assuming that p_interval is valid.
+	ERR_FAIL_INDEX_V_MSG(p_interval.idx, baked_point_cache.size(), Basis(), "Invalid interval");
+	if (up_vector_enabled) {
+		ERR_FAIL_INDEX_V_MSG(p_interval.idx, baked_up_vector_cache.size(), Basis(), "Invalid interval");
 	}
-	if (p_offset >= baked_max_ofs) {
-		return r[pc - 1];
+
+	int idx = p_interval.idx;
+	real_t frac = p_interval.frac;
+
+	Vector3 forward_begin;
+	Vector3 forward_end;
+	if (idx == 0) {
+		forward_begin = (baked_point_cache[1] - baked_point_cache[0]).normalized();
+		forward_end = (baked_point_cache[1] - baked_point_cache[0]).normalized();
+	} else {
+		forward_begin = (baked_point_cache[idx] - baked_point_cache[idx - 1]).normalized();
+		forward_end = (baked_point_cache[idx + 1] - baked_point_cache[idx]).normalized();
 	}
 
-	int start = 0;
-	int end = pc;
-	int idx = (end + start) / 2;
-	// Binary search to find baked points.
-	while (start < idx) {
-		real_t offset = baked_dist_cache[idx];
-		if (p_offset <= offset) {
-			end = idx;
-		} else {
-			start = idx;
-		}
-		idx = (end + start) / 2;
+	Vector3 up_begin;
+	Vector3 up_end;
+	if (up_vector_enabled) {
+		const Vector3 *up_ptr = baked_up_vector_cache.ptr();
+		up_begin = up_ptr[idx];
+		up_end = up_ptr[idx + 1];
+	} else {
+		up_begin = Vector3(0.0, 1.0, 0.0);
+		up_end = Vector3(0.0, 1.0, 0.0);
 	}
 
-	real_t offset_begin = baked_dist_cache[idx];
-	real_t offset_end = baked_dist_cache[idx + 1];
+	// Build frames at both ends of the interval, then interpolate.
+	const Basis frame_begin = Basis::looking_at(-forward_begin, up_begin);
+	const Basis frame_end = Basis::looking_at(-forward_end, up_end);
+	const Basis frame = frame_begin.lerp(frame_end, frac).orthonormalized();
```

Comment:
Wouldn't interpolation by `slerp()` instead of `lerp()` be more appropriate?

Reply:
but more expensive. can be changed if you want

Reply:
I recommend to use slerp for correct calculation. For performance, we have recently been preparing benchmark tests, so we can optimize later.

Reply:
done